src/auth/auth.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable } from '@nestjs/common';
+import { JwtService } from '@nestjs/jwt';
+import { RegisterDto } from './dto/register.dto';
+import { UserService } from 'src/user/user.service';
+import { RefreshDto } from './dto/refresh.dto';
+import {
+  NotFoundException,
+  UnauthorizedException,
+} from '@nestjs/common/exceptions';
+import { verify } from 'argon2';
+import { LoginDto } from './dto/login.dto';
+@Injectable()
+export class AuthService {
+  constructor(
+    private jwtService: JwtService,
+    private userService: UserService,
+  ) {}
+  async register(registerDto: RegisterDto) {
+    const user = await this.userService.create(registerDto);
+
+    const tokens = await this.issueTokens(user.id);
+
+    return {
+      user,
+      ...tokens,
+    };
+  }
+
+  async refresh(refreshDto: RefreshDto) {
+    const result = await this.jwtService.verifyAsync(refreshDto.refreshToken);
+
+    if (!result) {
+      throw new UnauthorizedException('Invalid refresh token');
+    }
+
+    const user = await this.userService.findUnique(result.id);
+
+    const tokens = await this.issueTokens(user.id);
```

Comment:
This one will create both new `accessToken` and `refreshToken`. I think `refresh` should create only a new `accessToken` and keep old `refreshToken`